### DISCUSSION TITLE
Burner Slider moves erratically

### DISCUSSION THREAD

**Discussion by danallen52** [artisan-roaster-scope/artisan#1081] [Q&A]
I am a new user of artisan, using V2.8.2.  I have completed setup using a Coffee-tech Ghibli 30Kilo ( touch)  and everything is communicating well.  The burner slider controls the burner on the roaster OK.  However after starting a roast, the burner displayed slider value moves erratically without user input;  The roaster burner setting does not change only the slider value displayed. For example, if the burner setting is 30, the displayed setting will show that for awhile then it may go to 10, then to 100, then to 0 without any pattern and the setting on the roaster remains at 30.  What can I do to correct this?

**Comment by poundy** [comment]
I am no expert in the codebase or the Ghibli roasters, but from a cursory review of the Ghibli Touch machine code ([here](https://github.com/artisan-roaster-scope/artisan/blob/master/src/includes/Machines/Coffee-Tech/Ghibli_Touch.aset)), there's nothing really here that Artisan is responsible for; how I interpret this is the value that the slider shows is derived from the machine, but a change in slider sends to the machine (and the value then will get amended as a new value is read).  In the first instance, I would always suggest that the first way to diagnose this form of communication problem is to replace cabling (and if you can describe how you have this all connected to us here, that would help visualise what you have). If it's USB, then you might need to look at an optically isolated USB cable.

**Comment by danallen52** [comment]
Here is my connection description:
Computer connected via cat6 cable direct  to router in roasting machine, and configured as per manufacturer instructions file as a modbus connection  to IP address of roaster per manufacturer:

![image](https://user-images.githubusercontent.com/66679442/214388269-7430b00e-bef5-4392-80ef-9f9ab646725f.png)

Slider and button setup configured by manufacturer setting file:

![Screenshot 2023-01-24 141653](https://user-images.githubusercontent.com/66679442/214387658-f6ae91cb-5e42-419f-9df4-2384508c0ea2.png)

unmodified slider event quantifier tab:

![image](https://user-images.githubusercontent.com/66679442/214387953-0b6f997d-f630-47e1-8294-8033890e403a.png)

**Reply by poundy** [reply]
OK. Two things spring to mind. 

As I mentioned, my first instinct is that there is some communication issue. Given your modbus-over-IP connection, that moves the potential place that the error exists from what I originally suspected (pc-to-roaster) to being PLC-to-sensor, within the roaster.  That means you'll need Coffe-Tech to help out here. Ask them if they've seen an issue like this and if they think there's any chance of a cabling or earthing issue with the burner sensor.

Second thought. It is possible that 2.8.2 has some nuance that is causing this. You could try an older version of Artisan, say 2.8.0 or 2.6.0, from the [releases page](https://github.com/artisan-roaster-scope/artisan/releases).  This would at least help eliminate recent changes that others may not have deployed yet, since 2.8.2 was only released a few days prior to Christmas and may not be in wide use yet.

**Comment by danallen52** [comment]
no solution yet.

**Reply by MAKOMO** [reply]
Could you please attach your Artisan settings here? To do this export them using menu Help >> Save Settings into a file named artisan-settings.aset. Rename that file to end with .txt and then drag-and-drop them to a message box here on this discussion thread on the Github Web page. It could be that you have the background replay event function turned on or something similar.

**Comment by danallen52** [comment]
[artisan-settings_t.txt](https://github.com/artisan-roaster-scope/artisan/files/10775088/artisan-settings_t.txt)
Thank you, Here is file with settings

**Reply by MAKOMO** [reply]
Under menu Config >> Events, 3rd tab "Sliders" untick the flag "Temp" for the burner entry (last). This is for sure wrong, but will not resolve your issue. Burner percentages are no temperatures that needs conversions between C and F.

<img width="1072" alt="Screenshot 2023-02-19 at 08 47 35" src="https://user-images.githubusercontent.com/13909011/219935779-643f7685-4956-46ce-8b69-000f52a59a74.png">

Under menu Config >> Events, 4th tab "Quantifier" untick the flag "Coarse".

<img width="1072" alt="Screenshot 2023-02-19 at 08 47 38" src="https://user-images.githubusercontent.com/13909011/219935781-f684cce5-8802-41c4-a178-f411dd56d78c.png">

If this not help, deactivate the quantifier entry for the burner completely. By doing this, data from the machine will not be able to move the slider any more. I couldn't see any other mechanisms active that would "move" your slider. it must be this quantifier mechanism.

**Reply by danallen52** [reply]
Marko, Thank you for your quick reply. I took the steps you suggested.  The deactivation of the quantifier check entry for burner stopped the slider issue.  Does deactivating the burner quantifier entry have any other effect that I should be aware of?  thank you, dan

**Reply by MAKOMO** [reply]
The Event Quantifier are responsible to have Artisan catch changes (here of the burner powere) on the machine using its touch display. If the quantifier for the burner is deactivated, Artisan will not be able to record such changes entered on the machine. However, changes on the Artisan side using its sliders or buttons will still be send to the machine.

**Reply by MAKOMO** [reply]
Dan, if you could attach here an Artisan roast profile during which such erratic slider moves happened or record a new sequence with your modified setup were you change the burner power using the machines touch input, I could take a look to see what data the machine is sending to Artisan and I could have a chance to find out why the quantifier is not working as it should in the first place.

Note that to attach a profile, you need to first rename the profile.alog file to profile.txt such that it ends with .txt and drag-and-drop it here to the comment box on the Github Web page.

**Reply by danallen52** [reply]
thank you, I will record one tomorrow and post it here... Sorry for the delay production did not allow time to experiment.  I will post ASAP, thanks for your patience.

**Reply by ryanhart** [reply]
I'm having the same problem with a Kaleido M6. The default settings work, but don't offer me the control I want. When I saved settings with some buttons removed and some added (I don't use the drum at low speeds but I do want 5% steps on the burner, for example), the sliders started moving erratically. The actual settings don't change, just the sliders. For example, drum=90%, burner=10%, air=0%, will say stable for a few minutes, but then burner and air will both jump to 45%. The fan isn't running and ET and BT aren't rising, so I know it's just the sliders.

I did check Quantifiers. Action was unchecked on all and I unchecked SV on all. I don't have Course, as in the screenshot above. No change.

I appreciate any ideas.
